shared/lib/multichain/chain-agnostic-permission.ts

```diff
@@ -146,6 +148,35 @@ export function getAllScopesFromCaip25CaveatValue(
   ) as CaipChainId[];
 }
 
+/**
+ * Gets all non-wallet namespaces from a CAIP-25 caveat value
+ * This extracts all namespaces from both required and optional scopes
+ * and returns a unique set.
+ *
+ * @param caip25CaveatValue - The CAIP-25 caveat value to extract namespaces from
+ * @returns Array of unique namespace strings
+ */
+export function getAllNonWalletNamespacesFromCaip25CaveatValue(
+  caip25CaveatValue: Caip25CaveatValue,
+): CaipNamespace[] {
+  return Array.from(
+    getAllScopesFromCaip25CaveatValue(caip25CaveatValue).reduce(
```

Comment:
nit. can we assign this value to a variable rather than inline it inside the `return Array.from()`?